pkg/module/custom_state_check.go

```diff
@@ -0,0 +1,67 @@
+package module
+
+import (
+	"fmt"
+	"github.com/kyma-project/lifecycle-manager/api/v1beta2"
+	"github.com/pkg/errors"
+)
+
+var validStates = []v1beta2.State{
+	v1beta2.StateReady,
+	v1beta2.StateProcessing,
+	v1beta2.StateError,
+	v1beta2.StateDeleting,
+	v1beta2.StateWarning,
+}
+
+var ErrCustomStateCheckValidation = errors.New("custom state check validation failed")
+
```

Comment:
Can you add some unit tests for all these new and exported functions. Should be a quick one :)

Reply:
Good Point. Added unit tests